### PR TITLE
Caching, styling per window, load on demand, system font support + tests

### PR DESCRIPTION
**Pull request by mikeger**
Hey
We are having some updates for Classy, based on our most recent usage experience:
- Styling per window
  It might be needed to specify extra windows where Classy should apply the style, or alternatively turn it off for main window. 
- Caching
  Our CAS file is taking about 500ms to parse, so we decided to cache it to binary plist using `NSCoding`, Classy supports now `id<CASCacheProtocol> cache` property to modify the caching behaviour
- Load on demand
  If app is starting on the background for performing the background tasks (APNS, VoIP, etc) we don't want nether to start Classy, nor to do any activity on views/windows
- System font support
  iOS 9 added `SanFrancisco` font family, which is hard to mention by name

Also contains tests for some of this functionality. Sorry for such a big PR.

### CODE REVIEW

**Review by dnedrow — COMMENTED** [review]
@mikeger What was the motivation behind bootStrapClassy? When I moved to v0.2.5 my styling stopped working until I called bootstrapClassy in my app delegate application:didFinishLaunchingWithOptions: method. Is that the correct place to call it?

File: Classy/Parser/CASStyler.h [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 @interface CASStyler : NSObject
 
-+ (void)bootstrapClassy;
++ (void)bootstrapClassyWithTargetWindows:(NSArray *)targetWindows;
```
Comment (dnedrow) [inline Classy/Parser/CASStyler.h]: @killobatt Why was this change made, and how should one bootstrap Classy now? IE., what windows are to be provided when calling this?

cc: @mikeger

**Comment by mikeger** [comment]
@dnedrow it's really nice to hear that someone is using Classy in production! 

Sorry for that change sneaking in, since the PR is from `master` to `master` it also included all other changes.

`bootStrapClassy` is useful when 
1. Your app sometimes does not need to load classy, for example when the app is started from the push notification on the background.
2. Classy needs to know the full list of windows in the application, if you have more than one.

Real-life use example can be found here: https://github.com/wireapp/wire-ios/blob/75f558d483324946ec846a7b600fef39db8fb09f/Wire-iOS/Sources/AppRootViewController.swift

**Comment by dnedrow** [comment]
@mikeger cool, thanks. So calling bootstrapClassy as below should be an acceptable setup if there is no need for special window handling?

        [CASStyler bootstrapClassyWithTargetWindows:UIApplication.sharedApplication.windows];

**Comment by mikeger** [comment]
Yes, if (1) code is not executed in the app extension (then there is no `UIApplication`) and (2) all your windows are loaded.

**Pull request merged**